File: src/javaFiles/Greeting.java
package javaFiles;

class Greeting {

    private String language;
    private int length;

    Greeting(String language) {
        this.language = language;
    }

    Greeting(String language, int length) {
        this.language = language;
        this.length = length;
    }

    void greeting() {
        System.out.printf("Hello: %s", this.language);
    }

    void loopGreeting() {
        for (int i =0; i< length; i++) {
            if (i == 0) {
                System.out.printf("%s: Hello: %s\n", i, this.language);
            }
        }
    }
}

File: src/ktFiles/Greeting.kt
package ktFiles

open class Greeting(private val language: String = "") {

    private var length = 0

    // Secondary constructor
    constructor(language: String, length: Int) : this(language) {
        this.length = length
    }

    fun greet() {
        println("Hello $language")
    }

    fun loopGreet() {
        for (i in 0..length) {
            if (i == 0) {
                println("Hello! $language")
            }
        }
    }
}